Common/src/main/java/com/blamejared/crafttweaker/api/util/sequence/SequenceContext.java

```diff
@@ -0,0 +1,51 @@
+package com.blamejared.crafttweaker.api.util.sequence;
+
+import com.blamejared.crafttweaker.api.annotation.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import org.openzen.zencode.java.ZenCodeType;
+
+/**
+ * A SequenceContext provides more data to {@link Sequence}s and allows a sequence to be stopped from inside a task.
+ */
+@ZenRegister
+@Document("vanilla/api/util/sequence/SequenceContext")
+@ZenCodeType.Name("crafttweaker.api.util.sequence.SequenceContext")
+public class SequenceContext<T, U> {
+    
+    private final Sequence<T, U> sequence;
+    private final U data;
+    
+    @ZenCodeType.Constructor
+    public SequenceContext(Sequence<T, U> sequence, U data) {
```

Comment:
Will this ctor ever be invoked from ZC?

Reply:
Someone *could* make a sequence the long way around, I can't think of any reason why they would, but they still could